corehq/toggles.py

```diff
@@ -265,6 +265,43 @@ def enabled(self, item, namespace=Ellipsis):
             or super(PredictablyRandomToggle, self).enabled(item, namespace)
         )
 
+
+class DynamicallyPredictablyRandomToggle(PredictablyRandomToggle):
+    """
+    A PredictablyRandomToggle whose randomness can be configured via the database/UI.
+    """
+    RANDOMNESS_KEY = 'hq_dynamic_randomness'
+
+    def __init__(
+        self,
+        slug,
+        label,
+        tag,
+        namespaces,
+        default_randomness=0,
+        help_link=None,
+        description=None,
+        always_disabled=None
+    ):
+        super(PredictablyRandomToggle, self).__init__(slug, label, tag, list(namespaces),
+                                                      help_link=help_link, description=description,
+                                                      always_disabled=always_disabled)
+        assert namespaces, 'namespaces must be defined!'
+        assert 0 <= default_randomness <= 1, 'default randomness must be between 0 and 1!'
+        self.default_randomness = default_randomness
+
+    @property
+    @quickcache(vary_on=['self.slug'])
+    def randomness(self):
+        from toggle.models import Toggle
```

Comment:
nit: can this be at the top of the file with other imports?

Reply:
good call. there was another inline import elsewhere in the file and i thought maybe someone had had to do that for a circular import situation, but looks like that's probably not the case.